libs/live-config/src/providers/jsonFileReader/index.ts

```diff
@@ -0,0 +1,33 @@
+import fs from "fs";
+import { Provider } from "..";
+import { ConfigInfo } from "../../LiveConfig";
+
+export class JsonFileReader implements Provider {
+  filePath: string;
+
+  constructor(config: { filePath: string }) {
+    if (!config.filePath.endsWith(".json")) {
+      throw new Error("Only .json files can be read by the provider");
+    }
+    this.filePath = config.filePath;
+  }
+
+  getValueBykey<K>(key: K, info: ConfigInfo) {
+    if (!fs.existsSync(this.filePath)) {
+      throw new Error("Invalid file path, could not read config");
+    }
+
+    try {
+      const fileContent = fs.readFileSync(this.filePath);
+      const parsedValue = JSON.parse(fileContent.toString());
+      return parsedValue[key] ?? info.default;
+    } catch (err) {
+      console.error(
+        `The config file doesn't have a key ${key}. Returning default value: ${info.default}`,
+        err,
+      );
+
+      return info.default;
```

Comment:
Shouldn't we manage the case where info is undefined/null ? Just a question

Accessing it directly typescript will warn us, but if we do something dynamic, for example accessing a family or currency dynamically, i'm not sure it will be fine

Maybe we should throw if info is null/undefined ?

Reply:
the jsonFileReader is not used in production. This is just an example to show how we add a new tailored provider in this lib.

We can handle the case for firebase provider when info is null/undefined 
please refer to
 /libs/live-config/src/providers/firebaseRemoteConfig/index.ts
and 
libs/live-config/src/providers/firebaseRemoteConfig/parser.ts

in the case of firebase provider that we use in production, when info is null/undefined, it means that when we use setConfig from our client. The injected config doesn't include that key. this is the case of featureflag. In my implementation, the setConfig only inject the cosmos family related config. So, when info is null/undefined, I don't throw an exception, instead, I handle it as an object and parse it with JSON.parse (see  /libs/live-config/src/providers/firebaseRemoteConfig/index.ts)

Reply:
Only used as a unit test